sdk/src/account_utils.rs

```diff
@@ -37,10 +39,32 @@ where
     }
 }
 
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum SystemAccountKind {
+    System,
+    Nonce,
+}
+
+pub fn account_is_system(account: &Account) -> Option<SystemAccountKind> {
```

Comment:
maybe should be named `get_account_kind()` ?

Reply:
and should probably be moved into runtime

Reply:
> maybe should be named `get_account_kind()` ?

How about `get_system_account_kind()`?  I'd rather not imply it should do something for non-system accounts.

> and should probably be moved into runtime

There wasn't an obvious home for it in runtime.  I'll look again

Reply:
system_instruction_processor?

Reply:
Sure!

Reply:
https://github.com/solana-labs/solana/pull/7645/commits/4c35fdd535def77c1971058eaf3e837e0c783b71